doc/source/tutorial.rst

```diff
@@ -0,0 +1,284 @@
+Tutorial
+========
+
+To use Ray, you need to understand the following:
+
+- How Ray uses object IDs to represent immutable remote objects.
+- How Ray executes tasks asynchronously to achieve parallelism.
+
+Overview
+--------
+
+Ray is a Python-based distributed execution engine. It can be used on a single
+machine to achieve efficient multiprocessing, and it can be used on a cluster
+for large computations.
+
+When using Ray, several processes are involved.
+
+- Multiple **worker** processes execute tasks and store results in object
```

Comment:
This is a good description of the system components, but I don't think this is the right place for it. This is the tutorial, so we should lead with the steps that a user themselves can take, and leave system description to later.